src/webext/checkDoNotTrack.ts

```diff
@@ -0,0 +1,9 @@
+export const doNotTrackSettingFromNavigatorIsActivated = () => {
```

Comment:
Some typescript details :
- add return type hint as the function is exported (seems even a good practice to me to always annotate function types)
- why not use a function instead of a variable assignation for a function declaration?
- I would simplify with `return doNotTrack === '1' || doNotTrack === 'yes'`, but it's really really minor

Reply:
> add return type hint as the function is exported (seems even a good practice to me to always annotate function types)

Will do. Can you give an example?

> why not use a function instead of a variable assignation for a function declaration?

What do you mean? Example?

> I would simplify with return doNotTrack === '1' || doNotTrack === 'yes', but it's really really minor

I prefer if statements for code readability. Not a big fan of || and ternary code for example but it's very subjective. If it's hurting the eyes too much for the team, I can change.

Reply:
> > add return type hint as the function is exported (seems even a good practice to me to always annotate function types)
> 
> Will do. Can you give an example?
> 

Sorry, yes of course.

```typescript
let sum = (x: number, y: number): number => {
    return x + y;
}
```

In your case, adding boolean:
```typescript
export const doNotTrackSettingFromNavigatorIsActivated = (): boolean => {
...
}
```

> > why not use a function instead of a variable assignation for a function declaration?
> 
> What do you mean? Example?

I mean using:

```typescript
export function doNotTrackSettingFromNavigatorIsActivated(): boolean {
}
```

instead of new let/const and arrow function syntax.

> 
> > I would simplify with return doNotTrack === '1' || doNotTrack === 'yes', but it's really really minor
> 
> I prefer if statements for code readability. Not a big fan of || and ternary code for example but it's very subjective. If it's hurting the eyes too much for the team, I can change.

In either proposition `||` is used, it was just about dropping some code (but no other modification).
Full example:

```typescript
export function doNotTrackSettingFromNavigatorIsActivated(): boolean { 
    const doNotTrack = window.doNotTrack || navigator.doNotTrack;
    return doNotTrack === '1' || doNotTrack === 'yes';
}
```